lean4-infoview/src/infoview/messages.tsx

```diff
@@ -81,15 +76,36 @@ export function MessagesAtFile({uri, messages}: MessagesAtFileProps) {
     );
 }
 
+function lazy<T>(f: () => T): () => T {
+    let state: {t: T} | undefined
+    return () => {
+        if (!state) state = {t: f()}
+        return state.t
+    }
+}
+
 /** Displays all messages for the specified file. Can be paused. */
 export function AllMessages({uri: uri0}: { uri: DocumentUri }) {
     const ec = React.useContext(EditorContext);
-    const dc = React.useContext(DiagnosticsContext);
+    const sv = React.useContext(VersionContext);
+    const rs = React.useContext(RpcContext);
+    const dc = React.useContext(LspDiagnosticsContext);
     const config = React.useContext(ConfigContext);
-    let diags0 = dc.get(uri0);
-    if (!diags0) diags0 = [];
+    const diags0 = dc.get(uri0) || [];
+
+    let iDiags0 = React.useMemo(() => lazy(async () => {
+        if (sv?.hasWidgetsV1()) {
+            try {
+                return await getInteractiveDiagnostics(rs, { uri: uri0, line: 0, character: 0 }) || [];
+            } catch (err: any) {
+                console.log('getInteractiveDiagnostics error ', err)
+            }
+        }
+        return diags0.map(d => ({ ...(d as LeanDiagnostic), message: { text: d.message } }));
+    }), [sv, rs, uri0, diags0]);
 
-    const [isPaused, setPaused, [uri, diags], _] = usePausableState(false, [uri0, diags0]);
+    const [isPaused, setPaused, [uri, diags, iDiags], _] = usePausableState(false, [uri0, diags0, iDiags0]);
+    if (isPaused) void iDiags()
```

Comment:
Is this fetching diagnostics when paused? Should it not do so on `!isPaused`?

Reply:
Yes, `iDiags` is a thunk and the code above forces it when the component switches from `!isPaused` to `isPaused`.  (It will also force it on every subsequent update when the component is paused, but those will be no-ops.)

The idea is that we don't need to load the diagnostics when the "all messages" view is closed and not paused.  But when it switches to paused, we need to fetch them.